### ISSUE TITLE
Tree widget: Keyboard navigation doesn't work in Models tree

### ISSUE THREAD

**Issue by grigasp** [iTwin/viewer-components-react#1032]
Repro:
1. Open test viewer.
2. In the tree widget's Models tree, click on the "show all" header button to focus it.
3. Click TAB key until the "apply filter" button is focused.
4. Click TAB one more time - notice the focus is gone:
   ![Image](https://github.com/user-attachments/assets/75d7cb47-d28f-4247-9a43-f664acf2a799)

Expected behavior: the next node is focused.

**Comment by saskliutas** [comment]
This works as designed. You can navigate tree with keyboard using arrows keys and use TAB key to reach interactable elements (checkbox, buttons) inside tree node.
